lms/static/scripts/frontend_apps/components/dashboard/test/ContentWithBadge-test.js

```diff
@@ -0,0 +1,31 @@
+import { checkAccessibility } from '@hypothesis/frontend-testing';
+import { mount } from 'enzyme';
+
+import ContentWithBadge from '../ContentWithBadge';
+
+describe('ContentWithBadge', () => {
+  function createComponent(props) {
+    return mount(<ContentWithBadge {...props} />);
+  }
+
+  [
+    { count: 15, hasMoreItems: false, expectedValue: '15' },
+    { count: 581, hasMoreItems: false, expectedValue: '581' },
+    { count: 200, hasMoreItems: true, expectedValue: '200+' },
+    { count: 100, hasMoreItems: true, expectedValue: '99+' },
```

Comment:
Perhaps worth adding a test case for `count: 1000` here. We don't currently special-case this, but a reader might expect us to given how we handle 100.